src/clean-core/string_stream.hh

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <cstring> // std::memcpy
+
+#include <clean-core/string.hh>
+#include <clean-core/string_view.hh>
+
+namespace cc
+{
+class string_stream
```

Comment:
ah just saw: in cc we use `struct` for value types like these. Also please add string_stream to `fwd.hh`